.gitignore

```diff
@@ -43,3 +43,8 @@
 *.un~
 Session.vim
 .netrwhist
+
+# Ignore vi backup files
```

Comment:
@pohly brought this up in the kubernetes-csi standup meeting.  We think actually the .gitignore should only include files/binaries that are generated by this repository and not for files dependent on a dev's environment.

So for this particular repo, I think we should only ignore the bin/ directory

Reply:
https://github.com/kubernetes-csi/external-provisioner/pull/204#discussion_r245847823